### DISCUSSION TITLE
Rejecting a task in a on-add hook in TW3

### DISCUSSION THREAD

**Discussion by meepleek** [GothenburgBitFactory/taskwarrior#3450] [Q&A] [answered]
How should tasks be rejected in TW3 when using an `on-add` hook?

This hook logs `nope`, but a task is still added. I believe this (returning an non-zero exit code) would not have been the case in TW2.
```fish
#!/usr/bin/env fish

echo Nope
exit 1
```

Diag:
```
task 3.0.2
   Platform: Linux

Compiler
    Version: 13.2.1 20230801
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++17

Build Features
     Commit: bc86a1e53
      CMake: 3.29.2
    libuuid: libuuid + uuid_unparse_lower
 Build type: 

Configuration
       File: /home/spc/.taskrc (found), 1671 bytes, mode 100644
       Data: /home/spc/.task (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
  rc.editor: hx
Hooks
     System: Enabled
   Location: /home/spc/.task/hooks
     Active: on-add-0_reject.fish (executable)
             on-add-validate-hook (executable) (symlink)
   Inactive: hook_config.toml     (not executable)unrecognized hook name

Tests
   Terminal: 233x62
       Dups: Scanned 317 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 317 tasks for broken references:
             No broken references found
```

**Comment by wrashi** [comment]
That's a good question I wish I knew how to answer, @SecretPocketCat! I would have expected that not passing anything would break the chain. Reading the [documentation](https://taskwarrior.org/docs/hooks_guide/), it seems like the reading in and printing a json dump just passes the values along.

```python
added_task = json.loads(sys.stdin.readline())
print(json.dumps(added_task))
```

This leads me to wonder if the receiving process doesn't receive a valid json object if it just ignores input from the hook and use the previous values. Not having done any hook work, that's only my best guess…

**Comment by djmitche** [comment, marked as answer]
Moved to https://github.com/GothenburgBitFactory/taskwarrior/issues/3455

**Reply by meepleek** [reply]
Thanks for confirming it's a bug. I've reverted to 2.6 for the time being.
